src/feature/flow_chart_panel/views/AppGalleryModal.tsx

```diff
@@ -18,62 +18,65 @@ import {
   SelectItem,
   SelectTrigger,
 } from "@/components/ui/select";
+import { Simulate } from "react-dom/test-utils";
+import error = Simulate.error;
 
 const subjectKeyList = ["fundamentals", "AI", "IO", "DSP"];
 const ignoreDir = [".github", "MANIFEST"];
 
-type AppGalleryModalProps = {
+export interface GithubJSON {
+  name: string;
+  path: string;
+  sha: string;
+  size: number;
+  url: string;
+  html_url: string;
+  git_url: string;
+  download_url?: string;
+  type: string;
+  links: object[];
+}
+
+interface AppGalleryModalProps {
   isGalleryOpen: boolean;
   setIsGalleryOpen: (open: boolean) => void;
-};
+}
 
 export const AppGalleryModal = ({
   isGalleryOpen,
   setIsGalleryOpen,
 }: AppGalleryModalProps) => {
-  const [selectFields, setSelect] = useState([]);
-  const [data, setData] = useState<object[]>([]);
-
-  useEffect(() => {
-    const fetchData = async () => {
-      const response = await fetch(
-        "https://api.github.com/repos/flojoy-ai/nodes/contents/?ref=main"
-      );
-      const raw = await response.json();
-      const filtered = raw.filter(
-        (obj) => obj["type"] === "dir" && !ignoreDir.includes(obj["name"])
-      );
-      setSelect(
-        filtered.map((obj) => (
-          <SelectItem key={obj.sha} value={obj.url}>
-            {obj.name}
-          </SelectItem>
-        ))
-      );
-    };
-    fetchData().catch(console.error);
-  }, []);
+  const [selectFields, setSelectFields] = useState<GithubJSON[]>([]);
+  const [searchData, setSearchData] = useState<GithubJSON[]>([]);
+  const [searchDisabled, setSearchDisabled] = useState<boolean>(true);
```

Comment:
This should be set to true again when you reopen the modal (just use a useEffect)